protocol/contracts/leaser/src/state/config.rs

```diff
@@ -67,10 +71,10 @@ impl Config {
             .map_err(Into::into)
     }
 
-    pub fn update_lease_code(storage: &mut dyn Storage, new_code: CodeId) -> ContractResult<()> {
+    pub fn update_lease_code(storage: &mut dyn Storage, new_code: Code) -> ContractResult<()> {
         Self::STORAGE
             .update(storage, |mut c| -> ContractResult<Config> {
-                c.lease_code_id = new_code;
+                c.lease_code = new_code;
```

Comment:
I'd suggest using functional style update. In my opinion, it could look a bit better especially in cases like the lines above in the `update` method.

```suggestion
                Ok(Self {
                    lease_code: new_code,
                    ..c
                })
```

Reply:
this is not in the scope of this PR, but I'll accept this suggestion, moreover prefer immutability.